src/tableaus.jl

```diff
@@ -78,28 +76,13 @@ function TableauRKExplicit(name::AbstractString, order::(@compat(Tuple{Vararg{In
                          convert(Matrix{T},b), convert(Vector{T},c) )
 end
 
-
-# TODO: remove conv_field
-conv_field{T,N}(D,a::Array{T,N}) = convert(Array{D,N}, a)
-
-
 lengthks(tab::TableauRKExplicit) = length(tab.c)
 
-# TODO: there should be a better way to do it
-function Base.convert{Tnew<:Real,T}(::Type{Tnew}, tab::TableauRKExplicit{T})
-    # Converts the tableau coefficients to the new type Tnew
-    newflds = ()
-    @compat for n in [:a,:b,:c]
-        fld = getfield(tab,n)
-        if eltype(fld)==T
-            newflds = tuple(newflds..., conv_field(Tnew, fld))
-        else
-            newflds = tuple(newflds..., fld)
-        end
-    end
-    TableauRKExplicit{Tnew}(tab.name, tab.order, newflds...) # TODO: could this be done more generically in a type-stable way?
-end
-
+Base.convert{Tnew<:Real,T}(::Type{Tnew}, tab::TableauRKExplicit{T}) =
+    TableauRKExplicit{Tnew}(tab.name, tab.order,
```

Comment:
shouldn't convert return the type it's converting to? this is a strange method, asking for a Real and getting back a tableau?